src/gui/tray/ActivityListModel.cpp

```diff
@@ -461,6 +465,28 @@ void ActivityListModel::triggerActionAtIndex(int id) const
     }
 }
 
+void ActivityListModel::handleAction(int activityIndex, int actionIndex)
+{
+    if (activityIndex < 0 || activityIndex >= _finalList.size()) {
+        qCWarning(lcActivity) << "Couldn't handle action on activity at index" << activityIndex << "/ final list size:" << _finalList.size();
+        return;
+    }
+
+    const auto actions = _finalList[activityIndex]._links;
+
+    if (actionIndex < 0 || actionIndex >= actions.size()) {
+        qCWarning(lcActivity) << "Couldn't handle action at index" << actionIndex << "/ actions list size:" << actions.size();
+        return;
+    }
+
+    if (actions[actionIndex]._verb == "WEB") {
+        QDesktopServices::openUrl(QUrl::fromUserInput(actions[actionIndex]._link));
+        return;
+    }
+
+    emit sendNotificationRequest(_finalList[activityIndex]._accName, actions[actionIndex]._link, actions[actionIndex]._verb, activityIndex);
```

Comment:
For readability reason I'd have an `activity` and an `action` intermediate variables introduced earlier, this would avoid all those `operator[]` calls throughout the method.